### PR TITLE
Refactor check_ssl.sh

### PR DESCRIPTION
**Pull request by tongpu**
A quick refactor of `check_ssl.sh` to simplify the code and remove some shellcheck warnings.

Changes have been verified to work with the following distributions:
- CentOS 7
- CentOS 8
- Ubuntu 18.04
- Ubuntu 20.04
- Debian 10

### CODE REVIEW

File: check_ssl.sh [hunk 1]
```diff
@@ -91,56 +91,43 @@ DATE_ACTUALLY_SECONDS=$(date +"%s")
...
-		# OpenSSL 1.0.X output format
-		COMMON_NAME=$(echo "${HOST_CHECK}" | grep "subject" | sed 's/^.*CN=//' | sed 's/\s.*$//')
-	elif [[ $(echo "${HOST_CHECK}" | grep "subject" | grep "CN = " > /dev/null; echo $?) -eq 0 ]]; then
-		# OpenSSL 1.1.X output format
-		COMMON_NAME=$(echo "${HOST_CHECK}" | grep "subject" | sed 's/^.*CN = //' | sed 's/\s.*$//')
-	fi
+	HOST_CHECK_COMMAND="openssl s_client -servername ${HOST} -connect ${IP}:${PORT}"
 fi
+HOST_CHECK=$(${HOST_CHECK_COMMAND} 2>&- | openssl x509 -enddate -subject -noout)
+while [ "${?}" = "1" ]; do
```
Comment (ayekat) [inline check_ssl.sh:103]: What's the purpose of using `while … do` here? Why not simply `if … then`?

File: check_ssl.sh [hunk 2]
```diff
@@ -3,9 +3,9 @@
 # HELP |
 #-------
 function HELP {
-NORM=`tput sgr0`
-BOLD=`tput bold`
-REV=`tput smso`
+NORM=$(tput sgr0)
+BOLD=$(tput bold)
+REV=$(tput smso)
 echo -e \\n"Help documentation for ${BOLD}$0${NORM}"\\n
```
Comment (ayekat) [inline check_ssl.sh]: As we're already in the business of refactoring, I would suggest replacing all these `echo` calls with a single heredoc into `cat`, i.e.

```bash
function HELP {
	…

	cat <<- EOF

	Help documentation for $BOLD$0$NORM
	$REV-H$NORM  -- Sets the value for...
	…
	EOF
}
```

**Review by ayekat — COMMENTED** [review]
The `getopts` loop at the beginning should probably also handle the cases `:` (missing argument for an option) and `'?'` (unknown option), to catch some user errors.

**--edit**: Ah, it appears there was already a `*` case for everything else. This catches the common issues, though a specific error message would also be nice instead of just "something is wrong".

File: check_ssl.sh (hunk 1 above)
Comment (tongpu) [reply check_ssl.sh:103]: Probably, but let's ask @luk43 why he implemented it that way.

File: check_ssl.sh (hunk 2 above)
Comment (tongpu) [reply check_ssl.sh]: If we go that far with the refactor I would proceed with making it POSIX compliant.

Please don't make me do that. :sob:

**Comment by tongpu** [comment]
@m-erhardt I've refactored your changes, so you're very welcome to provide feedback too.

File: check_ssl.sh (hunk 2 above)
Comment (ayekat) [reply check_ssl.sh]: I guess POSIX is a bit too much; we'd have to reimplement all the `read -p` as well :slightly_smiling_face:

I was mostly looking at this from a readability standpoint; at the very least, I would properly indent the function code by one level.

Comment (eyenx) [reply check_ssl.sh]: At this point, just rewrite it in go.

Comment (tongpu) [reply check_ssl.sh]: > At this point, just rewrite it in go.

Go away!

**Review by m-erhardt — APPROVED** [review]
Nice job combining the two sed regexp's for the CN. This is definitely more elegant than what I've done there before.

File: check_ssl.sh (hunk 2 above)
Comment (tongpu) [reply check_ssl.sh]: @ayekat I refactored it in 62df36bacf6735f8ad8f5b792f5c4bccff98d2d6, but somehow `cat <<-` didn't work and I had to use it without indentation.

Comment (ayekat) [reply check_ssl.sh]: The `<<-` variant only works if indented with a tab, not with spaces.

Given that the rest of the script uses tabs for indentation, I would propose (both for consistency's sake and for making the `<<-` notation work) to use tabs here as well.

Comment (tongpu) [reply check_ssl.sh]: Somehow my vim editorconfig integration refuse to indent the function with tabs? :shrug:

I indented with tabs now and it works. :tada:

File: check_ssl.sh (hunk 1 above)
Comment (tongpu) [reply check_ssl.sh:103]: I would say we can keep it like that for now.

**Pull request merged**